### ISSUE TITLE
viz-graph doesn't work if attribute values are something other than a string, keyword or number

### ISSUE THREAD

**Issue by standyck** [Engelberg/ubergraph#2]
It looks like the dorothy library calls its `escape-id` function on all attribute values as it is running the `dot*` function. If I have stored a record or a java object as an attribute value, the call to `d/dot` in `viz-graph` fails.  Here's a minimal illustrative example:

```
(uber/viz-graph (uber/graph [:a-node {:bad-attr (java.util.Date.)}]))
```

You might be able to use dorothy's `gen-id` function to fix it though I'm not sure what that function is doing exactly.

**Comment by Engelberg** [comment]
I'd be happy to fix this.  What do you think would be the best way for classes and records to print?  `str`, `pr-str`, `print-str` or something else?

**Comment by standyck** [comment]
My inclination is to say `print-str` since your just creating a graphic for human consumption. I guess it depends on what the output looks like. If you do something (silly, maybe, but ya never know) like

```
(uber/viz-graph (uber/graph [:a-node {:label [1 :a "b"]}]))
```

you'd want it to look like `[1 :a b]` on the graphic rather than `[1 :a \"b\"]`. I also can't think of any (sane) circumstances where the reader would ever need to consume the attribute map in there which argues against using `pr-str`. Does this make any sense?

**Comment by Engelberg** [comment]
I would agree with your argument of print-str over pr-str.

However, in your date example:

```
=> (str (java.util.Date.))
"Wed Jun 24 19:23:46 PDT 2015"
=> (print-str (java.util.Date.))
"#inst \"2015-06-25T02:24:02.075-00:00\""
```

In this case, the `str` version is arguably more readable.

But for records, you really want `print-str`, so maybe that's best.

**Comment by standyck** [comment]
And I can always override the `print-str` behavior for things like `java.util.Date` by implementing the `print-method` multimethod, right?

**Comment by Engelberg** [comment]
Fixed in v0.1.3

There were two aspects to the fix:
1. Make it so dorothy wouldn't choke because of a non number/string/keyword attribute in the attribute map.
2. Make it so the auto-label setting prints the attribute map in a reasonable way.  The auto-label function continues to use clojure.pprint to pretty-print the attribute map (mainly because it automatically handles splitting to multiple lines when the map gets large), with some new special handling for things that Dorothy had trouble with.

If you don't like the pretty-print behavior, you always have the option of putting in any `:label` you want for the node, simply by including your custom label in the attribute map.  Let me know how it works for you.

**Comment by standyck** [comment]
I upgraded and ran a few tests. It works great. Thanks for making the 
change and thanks for producing such a useful and well documented library!
